### ISSUE TITLE
[Feature request] Launch options/flags for connection settings

### ISSUE THREAD

**Issue by rdroimfd** [qTox/qTox#5077]
##### Brief Description

OS: Fedora 27
qTox version: (version numbers in menu `Settings → About`)  
Commit hash:  https://github.com/qTox/qTox/commit/5d0f9509a8d5fac940f402237efe4a8a6ab7e27b
toxcore:  0.2.1

##### Observed Behavior

At first I thought I had a bug that deleted settings, because I checked my advanced settings after updating and noticed my connection wasn't set to go through tor and had a random port instead of 9050. I eventually realized it was because I made a new account a few months ago and never set the same connection settings, and connection settings are saved on a per account basis.

I thought about ways this might be able to be changed, and the best one I could think of are launch options/cli flag settings to set connection settings, like proxy type/IP/UDP settings etc on launch.

Not only would this allow you to easily share this setting between accounts if you choose (such as changing your qTox shortcut to include them thus making any profile load those connection settings), but it would allow you to easily use your chosen settings on newly created accounts without ever having to connect to tox over a connection you don't want to.

What I mean is even if I had known I would need to set the connection options again after making a new account, when I first load into the new account, it would attempt to connect to the tox network over my default IP address until I go and manually change things (unless I disable my internet, load into the new profile, change to tor, and then reenable internet).

Edit: Was just reading over [this page](https://wiki.tox.chat/users/tox_over_tor_tot) and saw the example for doing it with toxic - " toxic –force-tcp –SOCKS5-proxy 127.0.0.1 9050"

**Comment by anthonybilinski** [comment]
>when I first load into the new account, it would attempt to connect to the tox network over my default IP address until I go and manually change things

This seems like the core problem. A more GUI solution would be to give users a "settings" option from the login window that would not connect to the Tox network, but still let them change profile settings. This would require some refactoring of qTox.

A third possibility would be to have an option to store settings globally instead of per-profile, where possible. We could save them in a global save file that all local profiles share, so that you don't need to re-set UI/network settings for each new profile you create. Users could toggle a profile between using the global or profile specific settings, and depending which one they select, their changes would only effect that set of settings.

Command line option is the easiest, but not the nicest longer term IMO.

**Comment by tox-user** [comment]
There are some settings in qTox that are global, even though they shouldn't. This should not be a global setting. What if someone wants to have multiple profiles for multiple things that use different proxies? Each profile should have their own network settings. We can add a separate option to configure global network settings that would work on all profiles and if someone wanted to, they could override their global setting with a local setting of their profile.

If we add a commandline option for this, we should also add one for using TCP mode. You need that to use Tor.

@rdroimfd you can use torsocks to always use Tox or any other program through Tor without having to rely on program's settings

**Comment by anthonybilinski** [comment]
>This should not be a global setting. What if someone wants to have multiple profiles for multiple things that use different proxies?

Agreed, all I'm suggesting is the option to have global settings, not to remove profile-specific settings. There are definitely cases where people want different network settings for different profiles, but I think there are also cases where people want the same GUI/network/privacy settings across their profiles, even though currently they're profile specific, so they have to keep re-setting the same options for each profile.

I'm thinking a tick box that says "use global settings", and if selected almost all settings are pulled from a global shared save, and your GUI settings forms now update the global settings save. If unchecked, almost all settings are pulled from your profile-specific settings, and GUI settings forms update your personal settings save.

**Comment by sudden6** [comment]
qTox now has network options to set via command line, @rdroimfd does this satisfy your needs?
